projects/core/src/occ/adapters/cms/occ-cms-component.adapter.ts

```diff
@@ -32,11 +40,23 @@ export class OccCmsComponentAdapter implements CmsComponentAdapter {
     id: string,
     pageContext: PageContext
   ): Observable<T> {
-    return this.http
-      .get<T>(this.getComponentEndPoint(id, pageContext), {
+    if (this.featureConfigService.isEnabled(USER_CMS_ENDPOINTS)) {
+      return this.userIdService.getUserId().pipe(
+        switchMap((userId: string) => {
+          return this.http.get<T>(
+            this.getComponentEndPoint(id, pageContext, userId),
+            {
+              headers: this.headers,
+            }
+          );
+        }),
+        this.converter.pipeable<any, T>(CMS_COMPONENT_NORMALIZER)
+      );
+    } else {
+      return this.http.get<T>(this.getComponentEndPoint(id, pageContext), {
```

Comment:
Let's move this return also outside else statement. Just for consistency.

Reply:
missed it, fixed now